tree/ntuple/v7/src/RCluster.cxx

```diff
@@ -0,0 +1,98 @@
+/// \file RCluster.cxx
+/// \ingroup NTuple ROOT7
+/// \author Jakob Blomer <jblomer@cern.ch>
+/// \date 2020-03-11
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2020, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RCluster.hxx>
+
+#include <TError.h>
+
+#include <iterator>
+#include <utility>
+
+
+ROOT::Experimental::Detail::ROnDiskPageMap::ROnDiskPageMap(ROnDiskPageMap &&other)
+   : fMemory(other.fMemory), fOnDiskPages(std::move(other.fOnDiskPages))
+{
+   other.fMemory = nullptr;
+   other.fOnDiskPages.clear();
+}
+
+
+ROOT::Experimental::Detail::ROnDiskPageMap &
+ROOT::Experimental::Detail::ROnDiskPageMap::operator =(ROnDiskPageMap &&other)
+{
+   fMemory = other.fMemory;
+   fOnDiskPages = std::move(other.fOnDiskPages);
+   other.fMemory = nullptr;
```

Comment:
Would it make sense to swap these two lines? (Handle `fMemory`, then `fOnDiskPages`?)